rfcs/2022-04-using-native-dialog.md

```diff
@@ -0,0 +1,102 @@
+
+The recommended implementation is to create one internal wrapping `<_Dialog>` component, which houses the `<dialog>` element and provides props that control styling and User interaction (i.e., if background content remains interactive). This component would also manage elevation and focus trapping behavior.
+
+Implementation of this component should not change the current component APIs for `<Drawer>` and `<Dialog>`.
+
+The reasoning behind creating one parent component is as follows:
+
+- Code is maintained in a singular file; DRY implementation
+- The `<dialog>` spec has a limited number of methods available, making the complexity of this component minimal
+  - `close()` will close the dialog
+  - `show()` will open the dialog, but background elements will remain interactive
+  - `showModal()` will open the dialog as a modal, background elements will **not** be interactive
+
+Simplified example of implementation:
+
+```js
+const _Dialog = (props) => {
+  <dialog className={customClass} {...dialogProps}>
+    {children}
+  </dialog>
+}
+
+const Drawer = (props) => {
+  <_Dialog className='ds-c-drawer' hasFocusTrap={bool}>
+    {/* Header */}
+    <Button onClose={handleOnClose}>
+    {/* Body */}
+    {/* Optional footer */}
+  </_Dialog>
+}
+
+const Dialog = (props) => {
+  <_Dialog className='ds-c-dialog'>
+    {/* Header */}
+    <Button onClose={handleOnClose}>
+    {/* Body */}
+    {/* Footer - Modal action buttons */}
+  </_Dialog>
+}
+```
+
+## Benefits
+
+By relying on the browser's implementation, the Design System stands to inherit several benefits:
+
+- Basic functional and accessibility enhancements become the browser's problem
+- Less reliance on third-parties, results in
+  - Publishing less code (better performance)
+  - A more flexible implementation (no longer beholden to someone else's API)
+  - Less Engineering time and effort spent updating
+
+## Risks
+
+Two risks have been identified with this approach:
+
+1. The Design System needs to implement a polyfill to continue IE support, which may result in less bundle size savings than initially suggested
+2. There may be specific implementations for `<Dialog>` and `<Drawer>` that weren't tested for in the initial discovery
+
+### 1. Polyfill support
+
+[Windows support of IE will end on June 15, 2022](https://docs.microsoft.com/en-us/lifecycle/announcements/internet-explorer-11-end-of-support). In alignment with other websites, and to encourage Users to update their browsers, I recommend removing all IE-specific polyfills following this date.
+
+**tldr; The addition of this polyfill is a temporary concern.**
+
+### 2. One-off bugs may be discovered
+
+The discovery of using the `<dialog>` element in the Design System was done using the pre-existing Storybook examples and documentation pages. As a result, there is a chance consuming teams have specific use-cases not documented in our system that this implementation will expose.
```

Comment:
Should it be mentioned that teams will have the ability to test these changes with our Beta release cycle to give them the opportunity to flag items early? It will be on teams to do this testing, but they have the opportunity to flag issues earlier rather than later